libs/server/Resp/RespServerSession.cs

```diff
@@ -336,6 +342,8 @@ private void ProcessMessages()
 
             var _origReadHead = readHead;
 
+            // Each message assumes the default behavior of committing the AOF log, given that AOF is enabled on the session
+            aofCommitCurrentCommand = true;
```

Comment:
If my understanding is correct, we are doing it at the message level.

1 message can have multiple commands. It should be done at the command level as the command can be mutating or non-mutating. But in current logic, if one of the commands in a message is mutating, it will wait to commit. It will not be helpful if it has an info command after a set command. It will block the info as well. I may be wrong but that's my understanding.

Reply:
this was initially on purpose, but I think now that I have changed my logic to only perform this  step on very lkimited set of commands we should be safe to move this to command level.